bokeh/core/properties.py

```diff
@@ -1255,6 +1263,23 @@ def from_json(self, json, models=None):
         else:
             raise DeserializationError("%s expected a dict or None, got %s" % (self, json))
 
+class ColumnData(Dict):
+    """Property holding column data in form of a dict. Also applies
+    encoding and decoding to the data.
+    """
```

Comment:
I think your suggestion of a ``ColumnArray`` property type is a good one. I'll go with that.